File: technocup/2019/elimination_round_1/d.triangle.py
n, m, k = map(int, input().split())
if 2*n*m % k:
    print('NO')
else:
    print('YES')
    print(0, 0)
    print(n, 1)
    print(n * ((2*m + k - 1) // k) - 2*n*m // k, (2*m + k - 1) // k)

File: technocup/2019/elimination_round_1/d.triangle.cpp
#include <iostream>

using namespace std;

int main()
{
	long long int n, m, k;
	cin >> n >> m >> k;
	if (2*n*m % k)
	{
		cout << "NO" << endl;
	}
	else
	{
		cout << "YES" << endl;
		cout << 0 << ' ' << 0 << endl;
		cout << n << ' ' << 1 << endl;
		cout << n * ((2*m + k - 1) / k) - 2*n*m / k << ' ' << (2*m + k - 1) / k << endl;
	}
	return 0;
}
